### PR TITLE
Create speed_conversions.py

### PR DESCRIPTION
**Pull request by Abinashbunty**
### Describe your change:
Code to convert speed from one unit to another 

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by CaedenPH — COMMENTED** [review]
First, you don't need two sets of the same doctests

Second, you may want to consider having a dictionary of each unit and instead of chaining lots of `if/elif`'s, you can just make dictionary lookups.

In addition, I don't think you should return the rounded speed, instead just leave it long

**Comment by Abinashbunty** [comment]
> In addition, I don't think you should return the rounded speed, instead just leave it long

I made the changes except for this one as this was producing too long results due to precision of `float` type. Is it fine? I rounded to 3 decimal places so that It looks neat.

**Comment by Abinashbunty** [comment]
@cclauss Is this fine? 😄

File: conversions/speed_conversions.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    >>> convert_speed(100, "mph", "knot")
+    86.898
+
+    >>> convert_speed(100, "knot", "km/h")
+    185.2
+
+    >>> convert_speed(100, "knot", "m/s")
+    51.444
+
```
Comment (cclauss) [inline conversions/speed_conversions.py:67]: Let's decrease the amount of scrolling required to read the implementation.  ;-)
```suggestion
    >>> convert_speed(100, "km/h", "mph")
    62.137
    >>> convert_speed(100, "km/h", "knot")
    53.996
    >>> convert_speed(100, "m/s", "km/h")
    360.0
    >>> convert_speed(100, "m/s", "mph")
    223.694
    >>> convert_speed(100, "m/s", "knot")
    194.384
    >>> convert_speed(100, "mph", "km/h")
    160.934
    >>> convert_speed(100, "mph", "m/s")
    44.704
    >>> convert_speed(100, "mph", "knot")
    86.898
    >>> convert_speed(100, "knot", "km/h")
    185.2
    >>> convert_speed(100, "knot", "m/s")
    51.444
```

File: conversions/speed_conversions.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+"""
+Convert speed units
+
+https://en.wikipedia.org/wiki/Kilometres_per_hour
+https://en.wikipedia.org/wiki/Miles_per_hour
+https://en.wikipedia.org/wiki/Knot_(unit)
+https://en.wikipedia.org/wiki/Metre_per_second
+
```
Comment (cclauss) [inline conversions/speed_conversions.py:8]:
```suggestion
```

File: conversions/speed_conversions.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    "km/h": 1.0,
+    "m/s": 0.277777778,
+    "mph": 0.621371192,
+    "knot": 0.539956803,
+}
+
+
+def convert_speed(speed: float, unit_from: str, unit_to: str) -> float:
+    """
+    Convert speed from one unit to another unit using the speed_chart
```
Comment (cclauss) [inline conversions/speed_conversions.py:28]:
```suggestion
    Convert speed from one unit to another using the speed_charts above.
```

File: conversions/speed_conversions.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    >>> convert_speed(100, "knot", "mph")
+    115.078
+    """
+    if unit_to not in speed_chart or unit_from not in speed_chart_inverse:
+        raise ValueError(
+            f"Incorrect 'from_type' or 'to_type' value: {unit_from!r}, {unit_to!r}\n"
+            f"Valid values are: {', '.join(speed_chart_inverse)}"
+        )
+
```
Comment (cclauss) [inline conversions/speed_conversions.py:76]:
```suggestion
```

File: conversions/volume_conversions.py [hunk 5]
```diff
@@ -52,11 +52,7 @@ def volume_conversion(value: float, from_type: str, to_type: str) -> float:
     0.000236588
     >>> volume_conversion(4, "wrongUnit", "litre")
     Traceback (most recent call last):
-      File "/usr/lib/python3.8/doctest.py", line 1336, in __run
-        exec(compile(example.source, filename, "single",
-      File "<doctest __main__.volume_conversion[7]>", line 1, in <module>
-        volume_conversion(4, "wrongUnit", "litre")
-      File "<string>", line 62, in volume_conversion
+    ...
```
Comment (cclauss) [inline conversions/volume_conversions.py:55]:
```suggestion
        ...
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: conversions/speed_conversions.py (hunk 2 above)
Comment (Abinashbunty) [reply conversions/speed_conversions.py]: Done

**Comment by Abinashbunty** [comment]
Now it's all set I hope. 😄 @cclauss

File: conversions/volume_conversions.py (hunk 5 above)
Comment (Abinashbunty) [reply conversions/volume_conversions.py:55]: Updated in PR #7116

**Comment by Abinashbunty** [comment]
@cclauss Now this can be merged. I resolved the conflict it was having due to another PR. 😄

**Review by cclauss — APPROVED** [review]

**Pull request merged**